pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -195,11 +194,12 @@ func UpdateImageTag(container *v1.Container, runtimeVersion *string, isvcConfig
 		return
 	}
 	if utils.IsGPUEnabled(container.Resources) && len(strings.Split(image, ":")) > 0 {
-		imageName := strings.Split(image, ":")[0]
-		if imageName == isvcConfig.Predictors.Tensorflow.ContainerImage {
-			container.Image = imageName + ":" + isvcConfig.Predictors.Tensorflow.DefaultGpuImageVersion
-		} else if imageName == isvcConfig.Predictors.PyTorch.ContainerImage {
-			container.Image = imageName + ":" + isvcConfig.Predictors.PyTorch.DefaultGpuImageVersion
+		// Can we detect the ServingRuntime here?
+		// For TFServing/TorchServe the GPU image is tagged with suffix "-gpu", when the version is found in the tag
+		// and runtimeVersion is not specified, we default to append the "-gpu" suffix to the image tag
+		re := regexp.MustCompile(`(:([\w.\-_]*))$`)
+		if len(re.FindString(image)) > 0 {
+			container.Image = image + "-gpu"
```

Comment:
For triton the image includes cuda libraries already, for tfserving and torchserve it has different images for cpu and gpu.